### PR TITLE
iOS fixes

### PR DESCRIPTION
**Pull request by BlueCobold**
A little more support for iOS. A minor fix for ContextSettings to handle stencil bits correctly and various changes to support orientation changes of the device. Tested on iOS 7.x and 8.x on emulators and real devices. Also includes retina-support. Both the retina-support and the orientation changes make use of the info.plist settings for the app for more freedom of the using developer.

### CODE REVIEW

**Comment by eXpl0it3r** [comment]
@LaurentGomila Anything else?

**Comment by LaurentGomila** [comment]
I haven't tested it myself, but it looks good.

**Comment by eXpl0it3r** [comment]
This PR has been added to my merge list, meaning it will be merged soon, unless someone raises any concerns.

File: src/SFML/Window/iOS/SFViewController.mm [hunk 1]
```diff
@@ -36,8 +36,6 @@ @implementation SFViewController
 ////////////////////////////////////////////////////////////
 - (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
 {
-    (void)interfaceOrientation;
```
Comment (mantognini) [inline src/SFML/Window/iOS/SFViewController.mm]: Don't you get a warning about unused variable here when that line is removed?

File: src/SFML/Window/iOS/SFAppDelegate.mm [hunk 2]
```diff
@@ -151,6 +162,36 @@ - (void)applicationWillTerminate:(UIApplication *)application
     }
 }
 
+- (bool)supportsOrientation:(UIDeviceOrientation)orientation
+{
+    if (!self.sfWindow)
```
Comment (mantognini) [inline src/SFML/Window/iOS/SFAppDelegate.mm]: That would be nitpicking but I'd had some empty lines to make this function more readable, especially after return statements. What do you guys think?

File: src/SFML/Window/iOS/SFAppDelegate.mm [hunk 3]
```diff
@@ -151,6 +162,36 @@ - (void)applicationWillTerminate:(UIApplication *)application
...
+    if ([supportedOrientations containsObject:@"UIInterfaceOrientationLandscapeLeft"])
+        appFlags += UIInterfaceOrientationMaskLandscapeLeft;
+    if ([supportedOrientations containsObject:@"UIInterfaceOrientationLandscapeRight"])
+        appFlags += UIInterfaceOrientationMaskLandscapeRight;
+    UIViewController* rootViewController = [((__bridge UIWindow*)(self.sfWindow->getSystemHandle())) rootViewController];
+    if (!rootViewController || ![rootViewController shouldAutorotate])
+        return false;
+    return (1 << orientation)
+        & [rootViewController supportedInterfaceOrientations]
+        & appFlags;
```
Comment (mantognini) [inline src/SFML/Window/iOS/SFAppDelegate.mm]: Is there a rationale for using `&` and not its cousin `&&`? If not I would stick to `&&` so we don't scratch our heads in the future wondering why it's `&` and not `&&` as the latter is used pretty much everywhere else for boolean expressions.

File: src/SFML/Window/iOS/SFView.mm [hunk 4]
```diff
@@ -170,16 +169,18 @@ - (void)layoutSubviews
...
 {
 	return [CAEAGLLayer class];
 }
 
-
 ////////////////////////////////////////////////////////////
--(id)initWithFrame:(CGRect)frame
+- (id)initWithFrame:(CGRect)frame andContentScaleFactor:(CGFloat)factor
 {
 	self = [super initWithFrame:frame];
```
Comment (mantognini) [inline src/SFML/Window/iOS/SFView.mm]: Tabs! :-P

**Comment by mantognini** [comment]
Here you got my nitpicking! =P

File: src/SFML/Window/iOS/SFViewController.mm (hunk 1 above)
Comment (BlueCobold) [inline src/SFML/Window/iOS/SFViewController.mm]: No, I didn't get any warning about it.

File: src/SFML/Window/iOS/SFAppDelegate.mm (hunk 2 above)
Comment (BlueCobold) [inline src/SFML/Window/iOS/SFAppDelegate.mm]: Empty lines after return statements? Never seen someone doing that, but that's what would stop this PR, its not a problem to add them.

File: src/SFML/Window/iOS/SFAppDelegate.mm (hunk 3 above)
Comment (BlueCobold) [inline src/SFML/Window/iOS/SFAppDelegate.mm]: It isn't a boolean expression, it is a bit-operation. && would give wrong results. I fact, using &&, the expression would always be true.

File: src/SFML/Window/iOS/SFView.mm (hunk 4 above)
Comment (BlueCobold) [inline src/SFML/Window/iOS/SFView.mm]: ?

File: src/SFML/Window/iOS/SFAppDelegate.mm (hunk 2 above)
Comment (LaurentGomila) [inline src/SFML/Window/iOS/SFAppDelegate.mm]: Yes. Don't hesitate to add empty lines in your code to make it more redable, by separating logical groups.

File: src/SFML/Window/iOS/SFView.mm (hunk 4 above)
Comment (LaurentGomila) [inline src/SFML/Window/iOS/SFView.mm]: We use spaces, not tabs.

File: src/SFML/Window/iOS/SFAppDelegate.mm (hunk 3 above)
Comment (mantognini) [inline src/SFML/Window/iOS/SFAppDelegate.mm]: My bad, a quick look at supportedInterfaceOrientations' doc shows that it's a bitmask so no change to be done here. (Our code style policy state that != 0 should not be used where an explicit conversion takes place, like here.)

File: src/SFML/Window/iOS/SFAppDelegate.mm (hunk 2 above)
Comment (BlueCobold) [inline src/SFML/Window/iOS/SFAppDelegate.mm]: I'll add some.

File: src/SFML/Window/iOS/SFAppDelegate.mm (hunk 3 above)
Comment (BlueCobold) [inline src/SFML/Window/iOS/SFAppDelegate.mm]: Its a style policy I fully agree with ;)

**Comment by eXpl0it3r** [comment]
@BlueCobold Any updates on this?

**Comment by BlueCobold** [comment]
@eXpl0it3r its up

**Comment by eXpl0it3r** [comment]
This PR has been added to my merge list, meaning it will be merged soon, unless someone raises any concerns.

**Comment by eXpl0it3r** [comment]
I squashed the third commit into the third commit.

Merged in cac4d58b3d009a95c41748f2aefbe489740c1798

**Pull request closed without merging**